hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -269,6 +284,160 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     }
   }
 
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    try {
+      String database = identifier.namespace().level(0);
+      String viewName = identifier.name();
+      Table table = clients.run(client -> client.getTable(database, viewName));
+      HiveViewOperations.validateTableIsIcebergView(
+          table, CatalogUtil.fullTableName(name, identifier));
+      clients.run(
+          client -> {
+            client.dropTable(database, viewName);
+            return null;
+          });
+      LOG.info("Dropped View: {}", identifier);
+      return true;
+
+    } catch (NoSuchViewException | NoSuchObjectException e) {
+      LOG.info("Skipping drop, View does not exist: {}", identifier, e);
+      return false;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + identifier, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropView", e);
+    }
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    Preconditions.checkArgument(
+        isValidateNamespace(namespace), "Missing database in namespace: %s", namespace);
+
+    try {
+      return listTablesByType(
+          namespace, TableType.VIRTUAL_VIEW, BaseMetastoreTableOperations.ICEBERG_VIEW_TYPE_VALUE);
+    } catch (UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all views under namespace " + namespace, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listViews", e);
+    }
+  }
+
+  private List<TableIdentifier> listTablesByType(
+      Namespace namespace, TableType tableType, String tableTypeProp)
+      throws TException, InterruptedException {
+    String database = namespace.level(0);
+    List<String> tableNames = clients.run(client -> client.getTables(database, "*", tableType));
+
+    // Retrieving the Table objects from HMS in batches to avoid OOM
+    List<TableIdentifier> filteredTableIdentifiers = Lists.newArrayList();
```

Comment:
This may be a good idea, but I feel it's a bit early now, we dont even do this for regular tables.

Can't we do this in a subsequent pr for both tables and views?

Reply:
What do you recommend here if we will use the same code of listTable it will be duplicate code again. Why not use the current approach for listView and use the new logic for listTable in next PR.

Reply:
On the previous [PR](https://github.com/apache/iceberg/pull/8907/files) , me and @pvary discussed about it and after the proposal we segregated this. Ref: https://github.com/apache/iceberg/pull/8907/files/477c7d1c24526e2983462c8fdd1f418264441aac#r1384740902 . It seems github has removed this link. 
Attaching image for reference. 
<img width="792" alt="Screenshot 2024-03-06 at 11 32 14 AM" src="https://github.com/apache/iceberg/assets/4146188/042629f0-8b0e-4589-a3f9-b07309545680">

Reply:
We do not keep all of the tables in memory. We only keep the `TableIndentifier`s in memory which is much smaller amount of data than the whole Hive `Table` object.

I have seen issues where there were many tables in a single database and it caused memory issues on HMS, and on the HMSClient side as well.